### PR TITLE
[open] new internal-fund-with-fee-rate flag

### PR DESCRIPTION
**Pull request by niteshbalusu11**
Added a new flag for bos open to bypass interaction which asks "Use internal funds?"

Files changed:
bos
peers/open_channels.js

### CODE REVIEW

**Comment by niteshbalusu11** [comment]
https://github.com/alexbosworth/balanceofsatoshis/issues/242

File: bos [hunk 1]
```diff
@@ -1054,6 +1054,7 @@ prog
   .option('--coop-close-address <addr>', 'Coop-close address', REPEATABLE)
   .option('--external-funding', 'Use external funds for the channel open')
   .option('--give <give_amount>', 'Amount to gift to peer', REPEATABLE)
+  .option('--internal-fund-with-fee-rate <internal_fee_rate>', 'Use internal funds and set fee rate', INT)
```
Comment (alexbosworth) [inline bos:1057]:
```suggestion
  .option('--internal-fund-at-fee-rate <pervbyte>', 'Use internal funding at chain fee rate', INT)
```

With set fee rate we have confusion with the ppm fee rate

File: peers/open_channels.js [hunk 2]
```diff
@@ -146,6 +146,10 @@ module.exports = (args, cbk) => {
           return cbk([400, 'MustSetOpeningNodeForEveryPublicKey']);
         }
 
+        if (!!args.is_external && !!args.internal_fund_fee_rate){
```
Comment (alexbosworth) [inline peers/open_channels.js:149]:
```suggestion
        if (!!args.is_external && !!args.internal_fund_fee_rate) {
```

File: peers/open_channels.js [hunk 3]
```diff
@@ -454,8 +462,8 @@ module.exports = (args, cbk) => {
 
       // Ask for the fee rate to use for internally funded opens
       askForFeeRate: ['isExternal', ({isExternal}, cbk) => {
-        // Exit early when there are no internal funds being spent
-        if (!!isExternal) {
+        // Exit early when there are no internal funds being spent or internal fee rate is specified
+        if (!!isExternal || !!args.internal_fund_fee_rate) {
```
Comment (alexbosworth) [inline peers/open_channels.js:466]: isExternal should already be checking for that internal fund so it doesn't need to be check again

Comment (niteshbalusu11) [reply peers/open_channels.js:466]: I guess I'll have to check here or return true on the ` isExternal:` function else it still asks for a funding rate, or you mean something else?

Comment (niteshbalusu11) [reply peers/open_channels.js:466]: If I return true in `isExternal`, then it'll ask for PSBT down below.

Comment (alexbosworth) [reply peers/open_channels.js:466]: Ah yeah ok that should be fine then

**Comment by niteshbalusu11** [comment]
Pushed changes.

**Comment by niteshbalusu11** [comment]
Closing, merged.

**Pull request closed without merging**